### ISSUE TITLE
support fig.alt for non HTML output

### ISSUE THREAD

**Issue by cderv** [yihui/knitr#1967]
This would be nice to have `fig.alt` work also for other outputs than Pandoc supports. 

See discussion and notes : 

* About support in DOCX: https://github.com/yihui/knitr/pull/1900#issuecomment-717101270
* About support in Pandoc in general: https://github.com/yihui/knitr/issues/1965#issuecomment-777584905

What we know is 

* Office supports alt text and Pandoc supports it. It would requires a special handling though as the syntax work well for docx only (not HTML for example)
* For latex and PDF, it does not seem to exist a generic way but some Journals seems to have solutions in there templates (For example : https://authors.acm.org/journals/how-to-write-alt-text-and-why). So maybe it could be possible to come up with a solution. 

Related question on community: 
* https://community.rstudio.com/t/how-to-include-both-alt-text-and-caption-in-a-figure-to-knit-to-word-with-rmarkdown/95662/2

**Comment by cderv** [comment]
See a user's workaround in https://github.com/rstudio/rmarkdown/issues/1867#issuecomment-801227405

**Comment by apreshill** [comment]
Hi @cderv (cc @hadley)

Saving this link here for reference on how to add alt text for MS outputs: https://support.microsoft.com/en-us/topic/add-alternative-text-to-a-shape-picture-chart-smartart-graphic-or-other-object-44989b2a-903c-4d9a-b742-6a75b451c669

**Comment by cderv** [comment]
See discussion in https://github.com/rstudio/rmarkdown/issues/2106#issuecomment-854486564 about PPTX format

**Comment by cderv** [comment]
For later, this is what we got from Pandoc when parsing the syntax

````powershell
❯ pandoc -t native
![look a puffin](https://upload.wikimedia.org/wikipedia/commons/9/94/Puffin_Mrkoww.jpg "Title: A proper puffin")
^Z
[Para [Image ("",[],[]) [Str "look",Space,Str "a",Space,Str "puffin"] ("https://upload.wikimedia.org/wikipedia/commons/9/94/Puffin_Mrkoww.jpg","fig:Title: A proper puffin")]]
````

**Comment by zachmayer** [comment]
I'd love to have alt text for markdown outputs too

**Comment by capnrefsmmat** [comment]
FYI, LaTeX is beginning to support alt text: the graphicx package now accepts an `alt` option to `\includegraphics`, as in

```latex
\includegraphics[width=3in,alt={University logo}]{logo.pdf}
```

and the [tagging project](https://github.com/latex3/tagging-project) is working on automatically embedding this in the PDF file for screenreaders to use. It'd be great if knitr supported passing `fig.alt` out to LaTeX, as the [impending accessibility rule](https://www.ada.gov/resources/2024-03-08-web-rule/) means many universities will have to use accessible PDFs only.

**Comment by cderv** [comment]
Thanks for sharing this news @capnrefsmmat ! 

Is tagged PDF already working with latest LaTeX builds ? 

Do you know if setting `alt=` attribute to older **graphicx** will error ? Does this require a version check for the available graphicx package ? 

BTW, markdown image is written in output, we'll need support for Pandoc for this. Maybe adding your comment to Pandoc's issue too
- https://github.com/jgm/pandoc/issues/6095

**Comment by capnrefsmmat** [comment]
Tagged PDF is currently experimental in LaTeX. It's opt-in (you have to add a new `\DocumentMetadata{}` command before the `\documentclass` to enable new PDF engine features) and not yet compatible with some packages. There's a status report from the LaTeX team here: <https://www.latex-project.org/publications/2024-FMi-UFi-TUB-tb140mittelbach-tagging.pdf>

It looks like their goal is for it to be mostly a drop-in system: add the extra command (which also specifies some document metadata and PDF format options) and it'll be a fully accessible, tagged PDF. There will be some complications with nonstandard packages, but once they have the core working, it should be possible to make the default knitr templates work with it.

The graphicx `alt` option was added in 2021. Judging by a quick test in Overleaf (which lets me select TeX Live 2020), versions from before 2021 will throw an error. So that's a hard compatibility limit. It might be possible to do a hack that checks the version and drops the `alt` option when it's too old. It'll already be necessary to do a version check to determine if `\DocumentMetadata` can be used, so that's maybe not too bad.

And thanks, I'll comment on the pandoc issue.

**Comment by capnrefsmmat** [comment]
In the mean time, should knitr support the `alt=` option in its TeX output formats, e.g. Rnw? I'm not familiar with the code, but it seems like that'd involve adding the option to [`hook_plot_tex()`](https://github.com/yihui/knitr/blob/c8f3a6db45e4382c306cd2eb841b7c552f594de0/R/hooks-latex.R#L183-L185), perhaps gated by a knitr option for users who have old TeX versions.

**Comment by yihui** [comment]
Yes.

**Comment by yihui** [comment]
`fig.alt` is now supported for the output formats where it can be expressed:

- **HTML** — the original support.
- **LaTeX/PDF** — added in #2378. knitr writes `\includegraphics[alt={...}]` directly (the `alt` option has been supported by **graphicx** since 2021). Together with the LaTeX tagging project's `\DocumentMetadata{}`, this feeds accessible tagged PDFs.

For the remaining formats discussed here — **DOCX/PPTX** — there is no good way to pass alt text through knitr's Markdown plot hook. Pandoc's Office writers *do* carry alt text (it becomes `wp:docPr@descr` in docx and `p:cNvPr@descr` in pptx), but they take it from the image's Markdown syntax `![alt](path)`, where that same text is also the figure caption. Markdown has no syntax to set alt text separately from the caption, so knitr cannot emit a distinct `fig.alt` alongside `fig.cap` — that would require writing raw OOXML (or a Lua filter) instead of Markdown. Hence knitr currently ignores `fig.alt` for Office output with a warning.

Given that LaTeX is done and Office alt text is not achievable through the Markdown hook, I'll close this. If there is demand for Office-specific alt text, a focused issue for a raw-OOXML/Lua-filter approach would be the way to go. Thanks all for the discussion and references.
